File: Anul 2/Semestrul 2/Metode de proiectare si programare/labs/Tema 4-proiect Server-Client Csharp/proiectSCcsharp/server/StartRpcServer.cs
using System;
using System.Collections.Generic;
using System.Configuration;
using proiectSCcsharp.repository;
using proiectSCcsharp.repository.interfaces;
using proiectSCcsharp.services;
using proiectSCcsharp.networking.utils;
using proiectSCcsharp.domain;
using proiectSCcsharp.server;
using proiectSCcsharp.domain.validators;

namespace proiectSCcharp.server
{
    public class StartRPCServer
    {
        [STAThread]
        private static string GetConnectionStringByName(string name)
        {
            string returnValue = null;
            var settings = ConfigurationManager.ConnectionStrings[name];
            if (settings != null)
                returnValue = settings.ConnectionString;
            return returnValue;
        }

        private static void Main(string[] args)
        {

            IDictionary<string, string> props = new SortedList<string, string>();
            props.Add("ConnectionString", GetConnectionStringByName("spectacoleDB"));
            Validator<Angajat> angajatValidator = new AngajatValidator();
            Validator<Spectacol> spectacolValidator = new SpectacolValidator();
            Validator<Bilet> biletValidator = new BiletValidator();
            RepoAngajat angajatRepository = new RepoAngajat(props);
            RepoSpectacol spectacolRepository = new RepoSpectacol(props);
            RepoBilet biletRepository = new RepoBilet(props);
            IService service = new Service(angajatValidator,spectacolValidator,biletValidator,angajatRepository,spectacolRepository,biletRepository);
            var server = new RpcConcurrentServer("127.0.0.1", 55556, service);
            try
            {
                server.Start();
                Console.WriteLine(@"Server started ...");
                Console.ReadLine();
            }
            catch (ServerException e)
            {
                Console.Error.WriteLine("Error starting the server" + e.Message);
            }
            finally
            {
                try
                {
                    server.Stop();
                }
                catch (ServerException e)
                {
                    Console.Error.WriteLine("Error stopping server " + e.Message);
                }
            }

        }
    }

}

File: Anul 2/Semestrul 2/MPP/labs/Tema 6-proiect Java ORM/proiectServerClientJava/Server/src/main/java/StartRpcServer.java

import proiectServerClientJava.domain.Angajat;
import proiectServerClientJava.domain.Spectacol;
import proiectServerClientJava.domain.Bilet;
import proiectServerClientJava.domain.validators.AngajatValidator;
import proiectServerClientJava.domain.validators.BiletValidator;
import proiectServerClientJava.domain.validators.SpectacolValidator;
import proiectServerClientJava.domain.validators.Validator;
import proiectServerClientJava.network.utils.AbstractServer;
import proiectServerClientJava.network.utils.RpcConcurrentServer;
import proiectServerClientJava.network.utils.ServerException;
import proiectServerClientJava.persistence.repository.jdbc.RepoAngajat;
import proiectServerClientJava.persistence.repository.jdbc.RepoAngajatORM;
import proiectServerClientJava.persistence.repository.jdbc.RepoSpectacol;
import proiectServerClientJava.persistence.repository.jdbc.RepoBilet;
import proiectServerClientJava.server.Service;
import proiectServerClientJava.service.IService;

import java.io.IOException;
import java.util.Properties;

public class StartRpcServer {
    private static int defaultPort=55555;

    public static void main(String[] args) {
        Properties serverProps=new Properties();
        try {
            serverProps.load(StartRpcServer.class.getResourceAsStream("/server.properties"));
            System.out.println("Server properties set. ");
            serverProps.list(System.out);
        } catch (IOException e) {
            System.err.println("Cannot find server.properties "+e);
            return;
        }
        Validator<Angajat> angajatValidator=new AngajatValidator();
        Validator<Spectacol> spectacolValidator=new SpectacolValidator();
        Validator<Bilet> biletValidator=new BiletValidator();
        RepoAngajatORM angajatRepository = new RepoAngajatORM();
        RepoSpectacol spectacolRepository = new RepoSpectacol(serverProps);
        RepoBilet biletRepository = new RepoBilet(serverProps);
        IService chatServerImpl=new Service(angajatValidator,spectacolValidator,biletValidator, angajatRepository, spectacolRepository, biletRepository);

        int chatServerPort=defaultPort;
        try {
            chatServerPort = Integer.parseInt(serverProps.getProperty("server.port"));
        }catch (NumberFormatException nef){
            System.err.println("Wrong  Port Number"+nef.getMessage());
            System.err.println("Using default port "+defaultPort);
        }
        System.out.println("Starting server on port: "+chatServerPort);
        AbstractServer server = new RpcConcurrentServer(chatServerPort, chatServerImpl);
        try {
            server.start();
        } catch (ServerException e) {
            System.err.println("Error starting the server" + e.getMessage());
        }finally {
            try {
                server.stop();
            }catch(ServerException e){
                System.err.println("Error stopping server "+e.getMessage());
            }
        }
    }
}